jquery.flot.hover.js

```diff
@@ -0,0 +1,303 @@
+            plot.hooks.bindEvents.push(bindEvents);
+            plot.hooks.shutdown.push(shutdown);
+            plot.hooks.drawOverlay.push(drawOverlay);
+        }
+
+        var triggerCleanupEvent = function(e) {
+            plot.unhighlight();
+            plot.getPlaceholder().trigger(new CustomEvent('plothovercleanup', { detail: e }));
+        }
+    }
+
+    function onMouseMove(e) {
+        var series = plot.getData();
+
+        if (options.grid.hoverable) {
+            triggerClickHoverEvent("plothover", e,
+                function(i) {
+                    return series[i]["hoverable"] !== false;
+                });
+        }
+    }
+
+    function onMouseLeave(e) {
+        if (options.grid.hoverable) {
+            triggerClickHoverEvent("plothover", e,
+                function(i) {
+                    return false;
+                });
+        }
+    }
+
+    function onClick(e) {
+        var series = plot.getData();
+
+        triggerClickHoverEvent("plotclick", e,
+            function(i) {
+                return series[i]["clickable"] !== false;
+            });
+    }
+
+    // trigger click or hover event (they send the same parameters
+    // so we share their code)
+    function triggerClickHoverEvent(eventname, event, seriesFilter, searchDistance) {
+        var options = plot.getOptions(),
+            offset = plot.offset(),
+            canvasX = event.pageX - offset.left,
+            canvasY = event.pageY - offset.top,
+            pos = plot.c2p({
+                left: canvasX,
+                top: canvasY
+            }),
+            distance = searchDistance !== undefined ? searchDistance : options.grid.mouseActiveRadius;
+
+        pos.pageX = event.pageX;
+        pos.pageY = event.pageY;
+
+        var item = plot.findNearbyItem(canvasX, canvasY, seriesFilter, distance);
+
+        if (item) {
+            // fill in mouse pos for any listeners out there
+            item.pageX = parseInt(item.series.xaxis.p2c(item.datapoint[0]) + offset.left, 10);
+            item.pageY = parseInt(item.series.yaxis.p2c(item.datapoint[1]) + offset.top, 10);
+        }
+
+        if (options.grid.autoHighlight) {
+            // clear auto-highlights
+            for (var i = 0; i < highlights.length; ++i) {
+                var h = highlights[i];
+                if ((h.auto === eventname &&
+                    !(item && h.series === item.series &&
+                        h.point[0] === item.datapoint[0] &&
+                        h.point[1] === item.datapoint[1])) || !item) {
+                    unhighlight(h.series, h.point);
+                }
+            }
+
+            if (item) {
+                highlight(item.series, item.datapoint, eventname);
+            }
+        }
+
+        plot.getPlaceholder().trigger(eventname, [pos, item]);
+    }
+
+    function highlight(s, point, auto) {
+        if (typeof s === "number") {
+            s = series[s];
```

Comment:
Where is the **series** variable coming from? Was this function copied from another function were the **series** variable was shared? Is this **if** statement really needed?

Reply:
Yes, this was copy-pasted. But I think we should move it inside init and keep the if statement. The highlight and unhighlight are part of the public plot API